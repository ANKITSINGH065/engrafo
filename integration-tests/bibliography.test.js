var utils = require("./utils");

test("a basic bibliography renders correctly", done => {
  utils.expectBodyToMatchSnapshot("bibliography/basic.tex", done);
});
