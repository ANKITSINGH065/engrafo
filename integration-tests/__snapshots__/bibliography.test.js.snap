// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`a 3-line inline bibliography renders correctly 1`] = `
<body>
  <dt-article
    class="centered"
  >
    <p>
      Refer to 
      <a
        href="#cite-1"
      >
        [1]
      </a>
      then refer to 
      <a
        href="#cite-2"
      >
        [2]
      </a>
      but misspell 
    </p>
  </dt-article>
  <dt-appendix>
    <div
      class="l-body"
    >
      <h3>
        References
      </h3>
      <dt-bibliography>
        <ol>
          <li
            id="cite-1"
          >
            <strong>
              Convolutional networks can learn to generate affinity graphs for image segmentation.
            </strong>
            <br />
            S. C. Turaga, J. F. Murray, V. Jain, F. Roth, M. Helmstaedter, K. Briggman, W. Denk, and H. S. Seung. 
            <em>
              Neural Comput.
            </em>
            , 22(2):511–538, Feb. 2010.
          </li>
          <li
            id="cite-2"
          >
            <strong>
              Segmentation fusion for connectomics.
            </strong>
            <br />
            A. Vazquez-Reina, M. Gelbart, D. Huang, J. Lichtman, E. Miller, and H. Pfister. In 
            <em>
              ICCV
            </em>
            , 2011.
          </li>
          <li
            id="cite-3"
          >
            <strong>
              Image segmentation by size-dependent single linkage clustering of a watershed basin graph.
            </strong>
            <br />
            A. Zlateski and H. S. Seung. 
            <em>
              CoRR
            </em>
            , abs/1505.00249, 2015.
          </li>
        </ol>
        <ol />
      </dt-bibliography>
    </div>
  </dt-appendix>
  <div
    id="cite-hover-boxes-container"
  />
  <div
    id="footnote-hover-boxes-container"
  />
</body>
`;

exports[`a basic bibliography renders correctly 1`] = `
<body>
  <dt-article
    class="centered"
  >
    <p>
      Hello.
      <dt-fn>
        <sup>
          <span
            id="fn-1"
            data-hover-ref="dt-fn-hover-box-0"
            style="cursor:pointer"
          >
            1
          </span>
        </sup>
      </dt-fn>
    </p>
  </dt-article>
  <dt-appendix>
    <div
      class="l-body"
    >
      <h3>
        Footnotes
      </h3>
      <dt-fn-list>
        <ol>
          <li>
            Just checking the bibliography comes after footnotes.
          </li>
        </ol>
      </dt-fn-list>
      <h3>
        References
      </h3>
      <dt-bibliography>
        <ol>
          <li
            id="cite-1"
          >
            <strong>
              Compositional morphology for word representations and language modelling.
            </strong>
            <br />
            Joe Bloggs and Phil Jones. 2014. In 
            <em>
              Proceedings of ICML
            </em>
            .
          </li>
          <li
            id="cite-2"
          >
            <strong>
              Morphological word-embeddings.
            </strong>
            <br />
            Ryan Cotterell and Hinrich Sch
            <span>
              ü
            </span>
            tze. 2015. In 
            <em>
              Proceedings of HLT-NAACL
            </em>
            .
          </li>
          <li
            id="cite-3"
          >
            <strong>
              Just a title, not a source.
            </strong>
            <br />
            Foo and Bar. 2020.
          </li>
        </ol>
        <ol />
      </dt-bibliography>
      <h3>
        Footnotes
      </h3>
      <dt-fn-list />
    </div>
  </dt-appendix>
  <div
    id="cite-hover-boxes-container"
  />
  <div
    id="footnote-hover-boxes-container"
  >
    <div
      style="display:none;"
      class="dt-hover-box"
      id="dt-fn-hover-box-0"
    >
      Just checking the bibliography comes after footnotes.
    </div>
  </div>
</body>
`;
